src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,372 @@
+     * @note If aMapName is larger than kMapNameMaxSize, it will be truncated.
+     */
+    MapStructureWrapper(uint8_t aMapId, const CharSpan & aMapName) { Set(aMapId, aMapName); }
+
+    /**
+     * @brief This is a copy constructor that initializes the map object with the values from another map object. All values are
+     * deep copied.
+     * @param[in] aOther The map object to copy.
+     */
+    MapStructureWrapper(const MapStructureWrapper & aOther) { *this = aOther; }
+
+    /**
+     * @brief This is an assignment operator that initializes the map object with the values from another map object. All values are
+     * deep copied.
+     * @param[in] aOther The map object to copy.
+     */
+    MapStructureWrapper & operator=(const MapStructureWrapper & aOther)
+    {
+        Set(aOther.mapID, aOther.name);
+        return *this;
+    }
+
+    /**
+     * @brief Set all fields of the mop object. All values are deep copied.
+     * @param[in] aMapId The identifier of this map.
+     * @param[in] aMapName A human readable name (should not be empty string).
+     *
+     * @note Requirements regarding what combinations of fields and values are 'valid' are not checked by this class.
+     * @note if aMapName is larger than kMapNameMaxSize, it will be truncated.
+     */
+    void Set(uint8_t aMapId, const CharSpan & aMapName)
+    {
+        mapID = aMapId;
+
+        // Save the map name to the buffer so that access and lifetime is controlled.
+        // Create a CharSpan reference the text.
+        // Make sure there is always a terminating character, mainly so we can get c_str for logging.
+        memset(mMapNameBuffer, 0, sizeof(mMapNameBuffer));
+
+        if (aMapName.size() == 0)
+        {
+            name = CharSpan(mMapNameBuffer, 0);
+        }
+        else if (aMapName.size() > sizeof(mMapNameBuffer))
+        {
+            // Save the truncated name that fits into available size.
+            memcpy(mMapNameBuffer, aMapName.data(), sizeof(mMapNameBuffer));
+            name = CharSpan(mMapNameBuffer, sizeof(mMapNameBuffer));
+        }
+        else
+        {
+            // Save full name.
+            memcpy(mMapNameBuffer, aMapName.data(), aMapName.size());
+            name = CharSpan(mMapNameBuffer, aMapName.size());
+        }
+    }
+
+    /**
+     * @brief Compare the map's name with given text.
+     * @param[in] aMapName The name to compare.
+     * @return true if the map structure's name field matches aMapName.
+     */
+    bool IsNameEqual(const CharSpan & aMapName) const { return name.data_equal(aMapName); }
+
+    /**
+     * @brief Get the map name (for logging).
+     * @return The map name as a c style char string.
+     *
+     * @note This is only available through the MapStructure object (no access from base structure).
+     */
+    const char * name_c_str() const { return mMapNameBuffer; }
+
+private:
+    // extra char to allow for terminating 0 (for name_c_str())
```

Comment:
But we don't terminate anywhere above....

Reply:
fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/2921b8a8deee83786ce4a6e447e78fb6d614e236).